api/logbroker.proto

```diff
@@ -80,6 +80,9 @@ message LogMessage {
 
 	// Data is the raw log message, as generated by the application.
 	bytes data = 4;
+	// Details is a map of key value pairs representing additional log details
+	// that may have been returned from the task logger
+	map<string, string> details = 5; 
```

Comment:
You're not wrong, but also in the docker daemon, details is represented by a map, and it doesn't seem to cause problems there.